rancher2/resource_rancher2_cluster.go

```diff
@@ -733,19 +744,12 @@ func getClusterKubeconfig(c *Config, id, origconfig string) (*managementClient.G
 				Actions: cluster.Actions,
 			}
 			err = client.APIBaseClient.Action(managementClient.ClusterType, action, clusterResource, nil, kubeConfig)
-			if err == nil {
-				if isRancher26 && len(token) > 0 {
-					newConfig, err := replaceKubeConfigToken(c, kubeConfig.Config, token)
-					if err != nil {
-						return nil, err
-					}
-					kubeConfig.Config = newConfig
+			if err != nil {
+				if !IsNotFound(err) && !IsForbidden(err) && !IsServiceUnavailableError(err) {
+					return nil, fmt.Errorf("Getting cluster Kubeconfig: %v", err)
 				}
-				return kubeConfig, nil
-			}
-			if !IsNotFound(err) && !IsForbidden(err) && !IsServiceUnavailableError(err) {
-				return nil, fmt.Errorf("Getting cluster Kubeconfig: %v", err)
 			}
+			return kubeConfig, nil
```

Comment:
@jakefhyde That seems to me like more of an enhancement and not part of this bug fix, and we've never seen an issue with generating a new kubeconfig.. unless you have? This block is also within a `context.Timeout` so it will wait up to the timeout for the HTTP response to resolve, I think.

Should I still update this logic to retry instead? WDYT?

Reply:
You have to do error handling, in the previous code, `return kubeConfig, nil` was inside of the `if err == nil` block. You moved it outside of the error handling, but the if statement does not have the same semantic meaning, since you removed the `if err == nil` branch. Now it will always return the kubeconfig, with or without an error, also dropping the error, in which case it is likely nil and will probably cause the provider to panic. Beforehand it would drop into the select statement, now it's unreachable code from that branch.